Tools/gdb/libpython.py

```diff
@@ -1034,14 +1034,18 @@ def __init__(self, gdbval):
         self._gdbval = gdbval
 
         if not self.is_optimized_out():
-            self.co = self._f_code()
-            self.co_name = self.co.pyop_field('co_name')
-            self.co_filename = self.co.pyop_field('co_filename')
-
-            self.f_lasti = self._f_lasti()
-            self.co_nlocals = int_from_int(self.co.field('co_nlocals'))
-            pnames = self.co.field('co_localsplusnames')
-            self.co_localsplusnames = PyTupleObjectPtr.from_pyobject_ptr(pnames)
+            try:
+                self.co = self._f_code()
```

Comment:
It would be better to get a clear signal right here and bail out as needed, than to have a blanket try/except around this entire section.

Reply:
Because this code runs inside gdb and there seems to be no way to see into the embedded Python interpreter, it is really hard to debug and experiment with. I'm reluctant to make any changes beyond the minimum to make it work.